torch/distributed/fsdp/_optim_utils.py

```diff
@@ -858,31 +858,26 @@ def _broadcast_unsharded_pos_dim_tensor_state(
 def _rekey_sharded_optim_state_dict(
     sharded_osd: Dict[str, Any],
     model: torch.nn.Module,
-    optim_input: Optional[
-        Union[
-            List[Dict[str, Any]],
-            Iterable[torch.nn.Parameter],
-        ]
-    ] = None,
+    optim: torch.optim.Optimizer,
```

Comment:
the previous code used to allow this not to be specified and by default use `list(model.parameters())`, is this something that we'd like to maintain?

Reply:
Resolved by preserving the existing code path.